frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/eviction-strategy-modal/eviction-strategy-modal.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { ModalFooter, ModalFooterProps } from '../../modal/modal-footer';
+import { withHandlePromise, HandlePromiseProps } from '@console/internal/components/utils';
+import { Checkbox, TextVariants, Text } from '@patternfly/react-core';
+
+import { k8sPatch } from '@console/internal/module/k8s';
+import { getEvictionStrategyPatch } from '../../../../k8s/patches/vm/vm-scheduling-patches';
+import { getVMLikeModel } from '../../../../selectors/vm/vmlike';
+
+import { VMLikeEntityKind } from '../../../../types/vmLike';
+
+import '../shared/scheduling-modals.scss';
+
+const EvictionStrategyModal: React.FC<EvictionStrategyModalProps> = withHandlePromise(
+  (props: EvictionStrategyModalWithPromiseProps) => {
+    const { vmLikeEntity, evictionStrategy, inProgress, close, handlePromise } = props;
+
+    const { t } = useTranslation();
+    const [isCheckedEvictionStrategy, setIsCheckedEvictionStrategy] = React.useState<boolean>(
+      !!evictionStrategy,
+    );
+
+    const onModalSubmit = () => {
+      if (isCheckedEvictionStrategy !== !!evictionStrategy) {
+        handlePromise(
+          k8sPatch(
+            getVMLikeModel(vmLikeEntity),
+            vmLikeEntity,
+            getEvictionStrategyPatch(vmLikeEntity, isCheckedEvictionStrategy),
+          ),
+          close,
+        );
+      }
+    };
+
+    const ModalFooterPropsObj: ModalFooterProps = {
```

Comment:
I don't see a reason why create an object for the props?
why not just pass it to the component?

Reply:
I think it make it more easy to add and remove props instead of creating a long line on the components itself, if it was 1-2 short props, so sure no need, but in this case i think i make it more readable.. 

But i can change if its not the standard :)